master/buildbot/test/integration/test_integration_secrets.py

```diff
@@ -0,0 +1,67 @@
+# This file is part of Buildbot.  Buildbot is free software: you can
+# redistribute it and/or modify it under the terms of the GNU General Public
+# License as published by the Free Software Foundation, version 2.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright Buildbot Team Members
+from __future__ import absolute_import
+from __future__ import print_function
+
+from twisted.internet import defer
+
+from buildbot.process.properties import Interpolate
+from buildbot.test.fake.secrets import FakeSecretStorage
+from buildbot.test.util.integration import RunMasterBase
+
+
+class SecretsConfig(RunMasterBase):
+
+    @defer.inlineCallbacks
+    def test_secret(self):
+        yield self.setupConfig(masterConfig())
+        build = yield self.doForceBuild(wantSteps=True, wantLogs=True)
+        self.assertEqual(build['buildid'], 1)
+        res = yield self.checkBuildStepLogExist(build, "echo bar")
+        self.assertTrue(res)
+
+    @defer.inlineCallbacks
+    def test_secretReconfig(self):
+        yield self.setupConfig(masterConfig())
+        self.reconfigMaster(masterConfig())
```

Comment:
following is enough:

    yield self.master.reconfig()

but you mush modify the config to change the secret or this test does not test much